replay_test.go

```diff
@@ -108,7 +108,8 @@ func TestValidReplayProvider(t *testing.T) {
 func TestFiniteReplayProvider(t *testing.T) {
 	t.Parallel()
 
-	p := &sse.FiniteReplayProvider{Count: 3}
+	p, err := sse.NewFiniteReplayProvider(3, false)
```

Comment:
Another test we should maybe do is to check that only the expected number of allocations is made – that is (including the suggestion above) a single allocation on creation.

This way it can be guaranteed that future updates to `FiniteReplayProvider` won't have memory issues and will display a consistent behavior.

[testing.AllocsPerRun](https://pkg.go.dev/testing#AllocsPerRun) or [testing.Benchmark](https://pkg.go.dev/testing#Benchmark) (+ adding the flag `-benchtime .01s` to CI tests so they don't take a century to complete) could be used for this. For now I'd pick the former and make a function which creates a `FiniteReplayProvider` with a small capacity (e.g. 3 messages), puts way more than 3 messages so we go over capacity and then replays to a [`mockClient`](https://github.com/tmaxmax/go-sse/blob/d2fe2fb42c5429bcdbfa13ab05b79c4cd4e3ae53/joe_test.go#L68-L71) which counts the messages to check (outside the `AllocsPerRun` function) that only 3 were replayed. The client and the messages should be created outside the function passed to `AllocsPerRun` so their allocations do not mess up the result.

This should test that no excess memory is allocated, that it doesn't indefinitely grow and that the number of events replayed doesn't unexpectedly change.